src/Tvheadend.cpp

```diff
@@ -418,16 +418,17 @@ PVR_ERROR CTvheadend::GetRecordings ( ADDON_HANDLE handle )
       PVR_RECORDING rec;
       memset(&rec, 0, sizeof(rec));
 
-      /* Channel name and icon */
+      /* Channel icon */
       if ((cit = m_channels.find(recording.GetChannel())) != m_channels.end())
       {
-        strncpy(rec.strChannelName, cit->second.GetName().c_str(),
-                sizeof(rec.strChannelName) - 1);
-
         strncpy(rec.strIconPath, cit->second.GetIcon().c_str(),
                 sizeof(rec.strIconPath) - 1);
       }
 
+      /* Channel name */
+      strncpy(rec.strChannelName, recording.GetChannelName().c_str(),
+              sizeof(rec.strChannelName) - 1);
```

Comment:
> what's wrong with an empty string as long it's initialized?

 Nothing. My fault, sorry. Mixed recording.GetChannelName() return value which can be empty with rec.strChannelName which cannot be zero lenght.